lib/cext/preprocess.rb

```diff
@@ -26,10 +26,18 @@
 		case Qfalse_int_const : mybool = 0; break;
 EOF
 
-PATCHED_FILES = {'xml_node_set.c'      => {:match =>  /[[:blank:]]*?switch\s*?\(.*?Qnil:/m, 
-                                           :replacement => XML_NODE_SET_PATCH},
-                 'pg_binary_encoder.c' => {:match => /[[:blank:]]*?switch\s*?\(.*?Qfalse\s*?:.*?break;/m,
-                                           :replacement => PG_BINARY_ENCODER_PATCH}}
+PATCHED_FILES = {'xml_node_set.c'      => [{:match =>  /[[:blank:]]*?switch\s*?\(.*?Qnil:/m, 
+                                           :replacement => XML_NODE_SET_PATCH}],
+                 'pg_binary_encoder.c' => [{:match => /[[:blank:]]*?switch\s*?\(.*?Qfalse\s*?:.*?break;/m,
+                                           :replacement => PG_BINARY_ENCODER_PATCH}],
+                 'bytebuffer.c'        => [{:match => /(static VALUE .*?) = Qnil;/,
+                                           :replacement => '\1;'}],
+                 'monitor.c'           => [{:match => /(static VALUE .*?) = Qnil;/,
```

Comment:
@eregon I've added a matcher to match again gem name. I was trying to keep it flexible to match against different possibilities like `ruby-pg/ext/pg_binary_encoder.c`, `pg-0.1.0/ext/pg_binary_encoder.c`, `ext/nokogiri/xml_node_set.c`, etc.